internal/fileupload/client.go

```diff
@@ -255,32 +154,56 @@ func (c *HTTPClient) flushBatch(ctx context.Context, revID RevisionID, batch *up
 		return fmt.Errorf("failed to upload files: %w", err)
 	}
 
-	// Close files in the uploaded batch
-	for _, file := range batch.files {
-		file.File.Close()
-	}
-
-	batch.totalUploaded += len(batch.files)
-	batch.reset()
 	return nil
 }
 
 // addPathsToRevision adds multiple file paths to an existing revision.
 func (c *HTTPClient) addPathsToRevision(ctx context.Context, revisionID RevisionID, rootPath string, pathsChan <-chan string, opts UploadOptions) error {
-	var chunks <-chan []string
+	fileSizeFilter := func(ff filterFile) filterOutcome {
+		fileSizeLimit := c.uploadRevisionSealableClient.GetLimits().FileSizeLimit
+		if ff.Stat.Size() > fileSizeLimit {
+			return filterOutcome{
+				shouldFilter: true,
+				reason:       uploadrevision.NewFileSizeLimitError(ff.Stat.Name(), ff.Stat.Size(), fileSizeLimit),
+			}
+		}
+		return filterOutcome{
+			shouldFilter: false,
+		}
+	}
 
-	if opts.SkipFiltering {
-		chunks = chunkChan(pathsChan, c.uploadRevisionSealableClient.GetLimits().FileCountLimit)
-	} else {
-		filter, err := c.createFileFilter(ctx)
+	fileNameLengthFilter := func(ff filterFile) filterOutcome {
```

Comment:
Should we rename this throughout to `pathLengthFilter` (including limits and various other variables)? I have no super strong opinion, but we are looking at the relative path and not just the filename.